src/super_gradients/examples/qat_from_recipe_example/qat_from_recipe.py

```diff
@@ -0,0 +1,239 @@
+
+    if "checkpoint_path" not in cfg.checkpoint_params:
+        raise ValueError("Starting checkpoint is a must for QAT finetuning.")
+
+    if cfg.quantization_params.modify_recipe_for_qat.enable:
+        modify_training_params_for_qat(cfg=cfg)
+
+    setup_device(
+        multi_gpu=core_utils.get_param(cfg, "multi_gpu", MultiGPUMode.OFF),
+        num_gpus=core_utils.get_param(cfg, "num_gpus"),
+        device="cuda",
+    )
+
+    train_dataloader = dataloaders.get(
+        name=cfg.train_dataloader,
+        dataset_params=cfg.dataset_params.train_dataset_params,
+        dataloader_params=cfg.dataset_params.train_dataloader_params.copy(),
+    )
+
+    val_dataloader = dataloaders.get(
+        name=cfg.val_dataloader,
+        dataset_params=cfg.dataset_params.val_dataset_params,
+        dataloader_params=cfg.dataset_params.val_dataloader_params.copy(),
+    )
+
+    # BUILD NETWORK
+    model = models.get(
+        model_name=cfg.arch_params.get("model_name", None) or cfg.architecture,
+        num_classes=cfg.get("num_classes", None) or cfg.arch_params.num_classes,
+        arch_params=cfg.arch_params,
+        pretrained_weights=cfg.checkpoint_params.pretrained_weights,
+        checkpoint_path=cfg.checkpoint_params.checkpoint_path,
+        load_backbone=cfg.checkpoint_params.load_backbone,
+    ).cuda()  # we assume that QAT is performed with CUDA
+
+    quantize_and_calibrate(
+        model,
+        train_dataloader,
+        method_w=cfg.quantization_params.selective_quantizer_params.method_w,
+        method_i=cfg.quantization_params.selective_quantizer_params.method_i,
+        per_channel=cfg.quantization_params.selective_quantizer_params.per_channel,
+        learn_amax=cfg.quantization_params.selective_quantizer_params.learn_amax,
+        skip_modules=cfg.quantization_params.selective_quantizer_params.skip_modules,
+        num_calib_batches=cfg.quantization_params.calib_params.num_calib_batches or (512 // cfg.dataset_params.train_dataloader_params.batch_size) or 1,
+        calibration_method=cfg.quantization_params.calib_params.calib_method,
+        percentile=cfg.quantization_params.calib_params.percentile,
+        verbose=cfg.quantization_params.calib_params.verbose,
+    )
+
+    logger.info("Performing validation of PTQ model...")
+
+    trainer = Trainer(experiment_name=cfg.experiment_name, ckpt_root_dir=get_param(cfg, "ckpt_root_dir", default_val=None))
+    val_results_tuple = trainer.test(model=model, test_loader=val_dataloader, test_metrics_list=cfg.training_hyperparams.valid_metrics_list)
+    valid_metrics_dict = get_metrics_dict(val_results_tuple, trainer.test_metrics, trainer.loss_logging_items_names)
+    results = ["Validate Results"]
+    results += [f"   - {metric:10}: {value}" for metric, value in valid_metrics_dict.items()]
+    logger.info("\n".join(results))
+
+    if cfg.training_hyperparams.max_epochs != 0:
```

Comment:
Im not sure I'm in favour of diffrentiating between PTQ and QAT on the basis of cfg.training_hyperparams.max_epochs ....
Can we make this more explicit ?